expression/builtin_cast.go

```diff
@@ -865,6 +865,15 @@ func (b *builtinCastRealAsDurationSig) evalDuration(row chunk.Row) (res types.Du
 		return res, isNull, err
 	}
 	res, err = types.ParseDuration(b.ctx.GetSessionVars().StmtCtx, strconv.FormatFloat(val, 'f', -1, 64), int8(b.tp.Decimal))
+	if err != nil {
```

Comment:
For reviewers, this error handling is copied from L1034/L1250, which is forgotten in former implementation.